File: lib/domain/usecases/branch/delete_branch_usecase.dart
import '../../repositories/i_branch_repository.dart';

/// 删除分支用例
class DeleteBranchUseCase {
  DeleteBranchUseCase(this._branchRepository);

  final IBranchRepository _branchRepository;

  Future<void> execute(String id) async {
    final branch = await _branchRepository.getById(id);
    if (branch == null) {
      throw ArgumentError('分支不存在: $id');
    }

    if (branch.isMain) {
      throw ArgumentError('不能删除主分支');
    }

    await _branchRepository.delete(id);
  }
}

File: web/src/application/usecases/branch/delete-branch-usecase.ts
import { IBranchRepository } from '../../../domain/repositories/i-branch-repository';

export class DeleteBranchUseCase {
  constructor(private branchRepo: IBranchRepository) {}

  async execute(id: string): Promise<void> {
    const branch = await this.branchRepo.getById(id);
    if (!branch) {
      throw new Error('分支不存在');
    }
    if (branch.isMain) {
      throw new Error('main 分支不能删除');
    }
    await this.branchRepo.delete(id);
  }
}
